library/src/messages/dapp-swarm-messages.ts

```diff
@@ -0,0 +1,138 @@
+import { SWARM_API_EVENT, SWARM_API_RESPONSE_EVENT } from '../constants/events'
+import { InpageReqMessageFormat, InterceptorResMessageFormat } from '../model/messages.model'
+import { getOnDocumentReadyPromise } from '../utils/window.util'
+import { SwarmMessages } from './swarm-messages'
+
+const MESSAGE_TIMEOUT = 30000
+/**
+ * Object that contains the resolve and reject functions of a promise
+ */
+class PromiseHandles {
+  private timeoutHandle: NodeJS.Timeout | null = null
```

Comment:
Actually there is a way to avoid explicitly defining the type. Instead this can be used: 
```typescript
private timeoutHandle: ReturnType<typeof setTimeout> | null = null
```